content for definitions

dalvarez83 opened the pull request:
**Created document for definitions for data science/AI toolkit**

** Next steps:**
- need to configure Figure 1 into document

dalvarez83 committed:
first commit of definitions content

dalvarez83 committed:
second commit for definitions.md

dalvarez83 committed:
third commit for definitions.md

dalvarez83 committed:
Merge branch 'main' of https://github.com/unicef/ooi-toolkit-ds into dalvarez-dstoolkit

merge branch to get icons

justwheel commented on content/01-definitions/definitions.md:
```diff
@@ -0,0 +1,24 @@
+
+Figure 1 diagram below depicts the AI ecosystem and the relationship between AI, ML, Deep Learning, NLP and RPA. 
+
+![Figure 1: The AI ecosystem](../../static/images/ai-ecosystem.png)
```
There are two ways to link images. I prefer the first method:

1. Upload the image into the git repository in a path that matches the name of the file, e.g. `static/01-definitions/definitions/ai-ecosystem.png`. Link using the file name (the suggestion offered in this comment).
2. Upload the file anywhere in the `static/` directory. All links must be hard-coded with paths, e.g. `/ooi-toolkit-ds/images/ai-ecosystem.png`.

```suggestion
![Figure 1: The AI ecosystem](ai-ecosystem.png)
```

dalvarez83 replied:
Thanks. I uploaded image to the path `static/images/01-definitions/definitions/ai-ecosystem.png` through a commit

dalvarez83 replied:
Should I create a separate pull request?

justwheel replied:
Looks like I gave you bad advice about where to store the files. I'll push a new commit to the Pull Request with the fix and explain it in a new comment.

justwheel commented on content/01-definitions/definitions.md:
```diff
@@ -0,0 +1,24 @@
+---
+title: "Definitions"
+icon: "ti-search"
```
This line you only need for an `_index.en.md` file. In the index file, you use that to initialize and create the category. Without an index page, your category will not appear on the site homepage. This page here will publish as `https://unicef.github.io/ooi-toolkit-ds/01-definitions/definitions/`

dalvarez83 replied:
Do you suggest I remove this descriptive annotation from the `definitions.md` file? Should I create a separate index file?

justwheel replied:
The description annotation is useful. This is used for OpenGraph data, or what pops up in the box when you share a link on social media or a messaging app. It should be a sentence or two summary of the article contents.

justwheel reviewed (commented):
This looks good to me for content. I shared a tip below on how to add images/figures into a document. You'll also need to upload the image to the pull request.

Also, this category won't appear on the site homepage without an index file, which might be needed before merging this. See in-line comment for more details.

dalvarez83 committed:
fourth commit on the content for definitions

dalvarez83 committed:
added ai-ecosystem.png

justwheel committed:
Change definitions page to index page, add image to category folder

This commit makes the following changes:

1. Change the `definitions.md` file to the category index page,
   `_index.en.md`. This causes the category to appear on the site home
   page, and it also makes this content the landing page. More content
   could be added in the future.
2. Move the `ai-ecosystem.png` file into the same directory as the
   category. This is how the URL inside the Markdown file can be most
   simple, and it also works with the Markdown view on GitHub too.
3. An extraneous `.gitkeep` file was removed since it is no longer
   needed for `content/`.

Signed-off-by: Justin W. Flory (he/him) [UNICEF Innovation] <jflory@unicef.org>

justwheel commented:
See my commit e0ebf6081062f7f79caafa3119ff36a422580d7d. I corrected my mistake about where the image needs to go; it should go in the same directory as the category where it will be used. That commit also gets the C.I. pipeline to pass, which was failing because the image link was incorrect.

@dalvarez83 Is this ready to merge?

dalvarez83 commented:
Appears ready to merge.

justwheel commented:
There was a bug with how the CI pipeline deployed the changes. I fixed it in commit d91c3ebb3528527460d9e7686857130af58452d7. The page is now published in the toolkit:

https://unicef.github.io/ooi-toolkit-ds/01-definitions/
